File: Wells/Source/Converters/DoubleStringConverter.vb
Imports System.Globalization
Imports Wells.Model

Public Class DoubleStringConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        If IsNumeric(value) Then
            Dim dbl = CType(value, Double)
            If dbl = BusinessObject.NullNumericValue Then
                Return "S/D"
            Else
                Return CDbl(value).ToString("N3")
            End If
        End If
        Return String.Empty
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        If IsNumeric(value) Then
            Dim str = CType(value, String)
            Dim doubleValue As Double
            Dim ok = Double.TryParse(str, NumberStyles.Any, CultureInfo.CurrentCulture, doubleValue)
            If ok Then
                Return doubleValue
            End If
        End If
        Return BusinessObject.NullNumericValue
    End Function
End Class

File: BaseView/src/Converters/DoubleStringConverter.cs
using Microsoft.VisualBasic;
using System;
using System.Globalization;
using System.Windows.Data;
using Wells.BaseModel.Models;
using Wells.Persistence.Repositories;

namespace Wells.BaseView.Converters
{
    public class DoubleStringConverter : IValueConverter
    {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (Information.IsNumeric(value))
            {
                var dbl = Convert.ToDouble(value, culture);
                if (double.Equals(dbl, BusinessObject.NumericNullValue))
                {
                    return "n/d";
                }
                if (parameter != null) { return dbl.ToString(parameter.ToString(), culture); }
                return dbl.ToString(culture);
            }
            return "0";
        }

        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (Information.IsNumeric(value))
            {
                var str = (string)value;
                bool ok = double.TryParse(str, NumberStyles.Any, culture, out double doubleValue);
                if (ok) { return doubleValue; }
            } 
            else if (value.ToString().ToLower().Trim() == "n/d")
            {
                return BusinessObject.NumericNullValue;
            }
            return 0.0;
        }
    }
}
